tests/include/test/psa_crypto_helpers.h

```diff
@@ -461,4 +461,26 @@ uint64_t mbedtls_test_parse_binary_string(data_t *bin_string);
 #define MBEDTLS_TEST_PSA_INTERNAL_KEYS          \
     MBEDTLS_TEST_PSA_INTERNAL_KEYS_FOR_DRBG
 
+/* Some helper macros to verify if MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE is
+ * large enough to contain an RSA key pair of the given size. This is meant to be
+ * used in test cases where MBEDTLS_PSA_STATIC_KEY_SLOTS is enabled. */
+#if defined(MBEDTLS_PSA_CRYPTO_C)
```

Comment:
Post CI run comment: a reason for having `MBEDTLS_TEST_STATIC_KEY_SLOTS_SUPPORT_RSA_[2048/4096]` guarded by `MBEDTLS_PSA_CRYPTO_CLIENT` is that `PSA_KEY_EXPORT_RSA_KEY_PAIR_MAX_SIZE()` is not defined when there's no PSA included in the build. This would cause tests like `test_no_psa_crypto_full_cmake_asan` to fail during the build because of errors like this:

```
tests/include/test/psa_crypto_helpers.h:467:49: error: function-like macro 'PSA_KEY_EXPORT_RSA_KEY_PAIR_MAX_SIZE' is not defined
#if (MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE >= PSA_KEY_EXPORT_RSA_KEY_PAIR_MAX_SIZE(4096))
                                                ^
```

Therefore I think I will keep the `MBEDTLS_PSA_CRYPTO_CLIENT` guard for these symbols.